src/main/scala/com/gu/support/workers/lambdas/FailureHandler.scala

```diff
@@ -71,4 +79,15 @@ class FailureHandler(emailService: EmailService)
   } yield result
 
   private def tryToDecode[T](errorJson: ErrorJson)(implicit decoder: Decoder[T]): Option[T] = decode[T](errorJson.errorMessage).toOption
+
+  def toCheckoutFailureReason(zuoraError: ZuoraError): CheckoutFailureReason = {
+    // Just get Stripe's decline code (example message from Zuora: "Transaction declined.do_not_honor - Your card was declined.")
+    val trimmedError = zuoraError.Message.stripPrefix("Transaction declined.").split(" ")(0)
```

Comment:
I take it this is the only way to get this code? It seems pretty nasty.

Reply:
Yes, unfortunately I think this is the only way to get it without making additional API calls (and that would become quite complicated)

Reply:
It's a shame that Zuora don't just return the decline_code directly from Stripe...